src/drivers/serac.cpp

```diff
@@ -107,24 +114,24 @@ int main(int argc, char *argv[])
   // Parse the arguments and check if they are good
   args.Parse();
   if (!args.Good()) {
-    if (myid == 0) {
+    if (rank == 0) {
       args.PrintUsage(std::cout);
     }
-    MPI_Finalize();
-    return 1;
+    serac::exit_gracefully(true);
   }
-  if (myid == 0) {
+  if (rank == 0) {
     args.PrintOptions(std::cout);
   }
 
   // Open the mesh
+  std::string msg = fmt::format("Opening mesh file: {0}", mesh_file);
+  SLIC_INFO_MASTER(rank, msg);
   std::ifstream imesh(mesh_file);
   if (!imesh) {
-    if (myid == 0) {
-      std::cerr << "\nCan not open mesh file: " << mesh_file << '\n' << std::endl;
-    }
-    MPI_Finalize();
-    return 2;
+    serac::flush_logger();
```

Comment:
Is there any way to get around manually flushing the logger on errors? I can see myself forgetting to do this.

Reply:
We can try playing around with it.  You need to be careful with the flushing because it has to be done collectively.  we could redefine SLIC_ERROR to do a flush inside of it. it just scares me a bit maybe unnecessarily.

Reply:
I think I prefer to have the flush in the error, but if that's not safe I can deal with it.

Reply:
I think we should merge this then ill work on duplicating the logging functions because that is going to require more thought and will be an easy SLIC_* to SERAC_* anywhere in the code.